packages/connect-node/src/http2-session-manager.ts

```diff
@@ -246,7 +246,7 @@ export class Http2SessionManager {
 
   private async gotoReady() {
     if (this.s.t == "ready") {
-      if (this.s.isShuttingDown()) {
+      if (this.s.conn.destroyed || this.s.conn.closed || this.s.isShuttingDown()) {
```

Comment:
This will avoid the problem found in https://github.com/bufbuild/connect-es/issues/683#issuecomment-1648209277. I think we can simply put _all_ connections that received any GOAWAY frame into the "shutting down" state. We will need a test.

Reply:
Pushed up a test in https://github.com/bufbuild/connect-es/pull/712/commits/bf4d7469304bcd4173dd05855fbcb5df31412ac8. New requests open a new connection 👍  

But the test framework reports an uncaught exception, see [run log](https://github.com/bufbuild/connect-es/actions/runs/5656284377/job/15323197715#step:5:201). That's because we remove event listeners for the connection here: https://github.com/bufbuild/connect-es/blob/4d1580fe8769a07be4aeb25f2eb4a411e90861f0/packages/connect-node/src/http2-session-manager.ts#L634-L644

Pushed up https://github.com/bufbuild/connect-es/pull/712/commits/a15680d5cba3187d5c72479a19131a9618119d70 to use isShuttingDown for all GOAWAY frames. This means we are deferring the cleanup in onExitState, and pass the test without uncaught exception.